### ISSUE TITLE
Build fails on OS X

### ISSUE THREAD

**Issue by edawson** [mklarqvist/tachyon#10]
I'm building tachyon on OS X using gcc 7.4 off the latest master branch, but it's failing to compile. The first error is here:

```
In file included from ./include/support_vcf.h:32:0,
                 from lib/support/support_vcf.cpp:1:
./include/buffer.h: In instantiation of 'void tachyon::SerializePrimitive(const T&, tachyon::yon_buffer_t&) [with T = long unsigned int]':
lib/support/support_vcf.cpp:206:36:   required from here
./include/buffer.h:161:9: error: ambiguous overload for 'operator+=' (operand types are 'tachyon::yon_buffer_t' and 'const long unsigned int')
  buffer += value;
```

There are then several errors of this type:
```
./include/buffer.h:166:9: error: cannot bind non-const lvalue reference of type 'double&' to an rvalue of type 'double'
  buffer >> value;
```
Is there a missing compiler flag perhaps? If it helps, most of the compiler errors seem to be in buffer.h.

**Comment by mklarqvist** [comment]
This is now fixed in the `dev` branch and will be merged shortly.

**Comment by mklarqvist** [comment]
Fixed in #12
